addons/website_event_sale/tests/test_frontend_buy_tickets.py

```diff
@@ -132,3 +119,80 @@ def test_buy_last_ticket(self):
     def test_pricelists_different_currencies(self):
         self.start_tour("/", 'event_sale_pricelists_different_currencies', login='admin')
     # TO DO - add public test with new address when convert to web.tour format.
+
+
+@odoo.tests.common.tagged('post_install', '-at_install')
+class TestRoutes(HttpCaseWithUserDemo, TestWebsiteEventSaleCommon, PaymentHttpCommon):
+
+    @mute_logger('odoo.http')
+    def test_check_seats_avail_before_purchase(self):
+        self.authenticate(None, None)
+
+        so_line_1, so_line_2 = self.env['sale.order.line'].create([
+            {
+                'event_id': self.event.id,
+                'event_ticket_id': self.ticket.id,
+                'name': self.event.name,
+                'order_id': self.so.id,
+                'product_id': self.ticket.product_id.id,
+                'product_uom_qty': 2,
+            },
+            {
+                'event_id': self.event_2.id,
+                'event_ticket_id': self.ticket_2.id,
+                'name': self.event_2.name,
+                'order_id': self.so.id,
+                'product_id': self.ticket_2.product_id.id,
+            },
+        ])
+        self.so._cart_update(line_id=so_line_1.id, product_id=self.ticket.product_id.id)
+        self.so._cart_update(line_id=so_line_2.id, product_id=self.ticket_2.product_id.id)
+        self.so.order_line.product_qty = 2
```

Comment:
This was needed because of this condition:
https://github.com/odoo/odoo/blob/0ce6c390183e80c75c3e41dfe4292f80af217174/addons/website_event_sale/models/sale_order.py#L27

While doing the flow manually I have the same state of the ordere line. I actually don't need to set the product_qty to 2 as well I'll push again.